apps/web/src/components/utils/hooks/useMessagePreviews.tsx

```diff
@@ -48,23 +55,51 @@ const useMessagePreviews = () => {
     return parsed.members.find((member) => member !== userProfileId) ?? null;
   };
 
+  useEffect(() => {
+    const updatePreviewMessages = async () => {
+      if (!client || !rawPreviewMessages) {
+        return;
+      }
+      const newPreviewMessages = new Map(previewMessages);
+      for (const msg of rawPreviewMessages) {
+        const existing = newPreviewMessages.get(msg.conversationKey);
+        // Only update the cache if the new messsage is newer
+        if (!existing || msg.sent > existing.sent) {
+          const message = await DecodedMessage.fromBytes(msg.messageBytes, client);
+          const { conversationKey } = msg;
+          newPreviewMessages.set(conversationKey, message);
+        }
+        setPreviewMessages(newPreviewMessages);
+      }
+    };
+
+    updatePreviewMessages();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [client, rawPreviewMessages]);
+
   useEffect(() => {
     if (profilesLoading) {
       return;
     }
     const toQuery = new Set(profileIds);
-    // Don't both querying for already seen profiles
-    for (const profile of Array.from(messageProfiles.values())) {
-      toQuery.delete(profile.id);
+    if (messageProfiles) {
+      // Don't both querying for already seen profiles
+      for (const profile of Array.from(messageProfiles.values())) {
+        toQuery.delete(profile.id);
+      }
     }
 
     if (!toQuery.size) {
       return;
     }
 
     const loadLatest = async () => {
-      setProfilesLoading(true);
-      const newMessageProfiles = new Map(messageProfiles);
+      // The DB has been loaded, but there are no results in it.
+      if (messageProfiles && !messageProfiles?.size) {
+        console.log('No message profiles');
+        setProfilesLoading(true);
```

Comment:
This is hacky as hell, but only sets the loading status to `true` when there is a clean cache.